### ISSUE TITLE
Tapping on PlainText powered blocks does not make the block-holder selected

### ISSUE THREAD

**Issue by mzorz** [wordpress-mobile/gutenberg-mobile#251]
What I want now is to be able to tap into a PlainText block, see the cursor appears, and make the Block automatically get selected (bubble the event up to get `isSelected` to be set to `true`).
Currently, what happens is you have to tap on the borders of the block to make it be selected.

This happens for PlainText-based blocks as well, so it should be easier to isolate the problem by not using RecyclerView nor Aztec to try and understand what's going on.

Ideas:
- make the touch event be grabbed first by the touchable area in the block-holder? (I mean, investigate around that)
- if that doesn't work, then using `onFocus` in PlainText to bubble up a `Selected` event, that makes `isSelected` turn to true for the block holder representing such a block.
- furthermore, there are plans to use InputTextState-like state, but that's on a different level (keyboard handling will need this to be able to hide/show the keyboard not only on InputText but also on Aztec RN wrapper).

**Comment by mzorz** [comment]
Possiby related issue https://github.com/facebook/react-native/issues/14958#issuecomment-324237317

**Comment by mzorz** [comment]
> furthermore, there are plans to use InputTextState-like state, but that's on a different level (keyboard handling will need this to be able to hide/show the keyboard not only on InputText but also on Aztec RN wrapper).

Plans referred to up here are exposed in #273.

**Comment by daniloercoli** [comment]
It seems that the problem is not only related to PlainText, since i'm experiencing it on Rich Text fields on a real device.

I've just put some logs in the code, and moving the caret from a Para block to a Code blocks doesn't update the props properly.
See the name of the component that holds the focus after the tapping, during the rendering phase:
```
2018-11-23 19:12:26.696 28060-28100/? I/ReactNativeJS: 1542996746696
2018-11-23 19:12:26.696 28060-28100/? I/ReactNativeJS: core/image
2018-11-23 19:12:26.696 28060-28100/? I/ReactNativeJS: NOT FOCUSED
2018-11-23 19:12:26.733 28060-28100/? I/ReactNativeJS: 1542996746733
2018-11-23 19:12:26.733 28060-28100/? I/ReactNativeJS: core/image
2018-11-23 19:12:26.734 28060-28100/? I/ReactNativeJS: NOT FOCUSED
2018-11-23 19:12:26.750 28060-28100/? I/ReactNativeJS: 1542996746750
2018-11-23 19:12:26.751 28060-28100/? I/ReactNativeJS: gmobile/unsupported
2018-11-23 19:12:26.751 28060-28100/? I/ReactNativeJS: NOT FOCUSED
2018-11-23 19:12:26.761 28060-28100/? I/ReactNativeJS: 1542996746761
2018-11-23 19:12:26.761 28060-28100/? I/ReactNativeJS: core/heading
2018-11-23 19:12:26.761 28060-28100/? I/ReactNativeJS: NOT FOCUSED
2018-11-23 19:12:26.774 28060-28100/? I/ReactNativeJS: 1542996746774
2018-11-23 19:12:26.775 28060-28100/? I/ReactNativeJS: core/paragraph
2018-11-23 19:12:26.775 28060-28100/? I/ReactNativeJS: NOT FOCUSED
2018-11-23 19:12:26.787 28060-28100/? I/ReactNativeJS: 1542996746786
2018-11-23 19:12:26.787 28060-28100/? I/ReactNativeJS: core/nextpage
2018-11-23 19:12:26.787 28060-28100/? I/ReactNativeJS: NOT FOCUSED
2018-11-23 19:12:26.823 28060-28100/? I/ReactNativeJS: 1542996746822
2018-11-23 19:12:26.825 28060-28100/? I/ReactNativeJS: core/paragraph
2018-11-23 19:12:26.825 28060-28100/? I/ReactNativeJS: FOCUSED
2018-11-23 19:12:26.910 28060-28100/? I/ReactNativeJS: 1542996746910
2018-11-23 19:12:26.910 28060-28100/? I/ReactNativeJS: core/code
2018-11-23 19:12:26.911 28060-28100/? I/ReactNativeJS: NOT FOCUSED
2018-11-23 19:12:26.944 28060-28100/? I/ReactNativeJS: 1542996746944
2018-11-23 19:12:26.944 28060-28100/? I/ReactNativeJS: core/more
2018-11-23 19:12:26.944 28060-28100/? I/ReactNativeJS: NOT FOCUSED
2018-11-23 19:12:26.976 28060-28100/? I/ReactNativeJS: 1542996746976
2018-11-23 19:12:26.976 28060-28100/? I/ReactNativeJS: gmobile/unsupported
2018-11-23 19:12:26.976 28060-28100/? I/ReactNativeJS: NOT FOCUSED
```

The `core/paragraph` block that had the focus previously still has its prop `this.props.focused ` set to true. The other `core/code` block, the one in which I moved the caret, doesn't have it set to true.

**Comment by mzorz** [comment]
Closed via #277
